Review and add new linters to improve the code quality of telemetry manager. (#112)
kyma-project/telemetry-manager: docs/linter.md

# Sourcecode linting

We use [golangci-lint](https://golangci-lint.run) with fine-grained configuration for the source code linting.

## Linters in action

> Below is the list of linters configured and integrated as a CI stage through a [ProwJob](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/components/kyma-components-static-checks.yaml#L6).

Reviewer: ```suggestion
The following linters are configured and integrated as a CI stage through a [ProwJob](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/components/kyma-components-static-checks.yaml#L6).
```

| [`errcheck`](https://github.com/kisielk/errcheck) [⛭](https://golangci-lint.run/usage/linters/#errcheck)                                                          | checks for unhandled errors                                      | [inline //nolint](#inline-nolint) |
| [`errchkjson`](https://github.com/breml/errchkjson) [⛭](https://golangci-lint.run/usage/linters/#errchkjson)                                                      | checks types passed to the json encoding functions               | [inline //nolint](#inline-nolint) |
| `exportloopref`                                                                                                                                                   | finds exporting pointers for loop variables                      | [inline //nolint](#inline-nolint) |
| [`gci`](https://github.com/daixiang0/gci) [⛭](https://golangci-lint.run/usage/linters/#gci)                                                                       | checks import order and ensures it is always deterministic       | [inline //nolint](#inline-nolint) |

> | [`ginkgolinter`](https://github.com/nunnatsa/ginkgolinter) [⛭](https://golangci-lint.run/usage/linters/#ginkgolinter)                                             | enforces standards of using ginkgo and gomega                    | [inline //nolint](#inline-nolint) |

Reviewer: ```suggestion
| [`ginkgolinter`](https://github.com/nunnatsa/ginkgolinter) [⛭](https://golangci-lint.run/usage/linters/#ginkgolinter)                                             | enforces standards of using Ginkgo and Gomega                    | [inline //nolint](#inline-nolint) |
```
